src/PointwiseFunctions/AnalyticSolutions/NewtonianEulerSolutions/IsentropicVortex.hpp

```diff
@@ -0,0 +1,175 @@
+class er;
+}  // namespace PUP
+
+namespace NewtonianEuler {
+template <typename DataType>
+struct MassDensity;
+template <typename DataType, size_t Dim, typename VolumeFrame = Frame::Inertial>
+struct Velocity;
+template <typename DataType>
+struct SpecificInternalEnergy;
+}  // namespace NewtonianEuler
+/// \endcond
+
+// IWYU pragma: no_forward_declare Velocity
+
+namespace NewtonianEuler {
+namespace Solutions {
+
+/*!
+ * \ingroup NewtonianEulerSolutionsGroup
+ * \brief Newtonian isentropic vortex in Cartesian coordinates
+ *
+ * The well known analytic solution to the 2-D Newtonian Euler system
+ * representing the slow advection of an incompressible, isentropic vortex [1].
+ * The initial condition is the superposition of a mean uniform flow and a small
+ * fluid perturbation of gaussian profile and given strength. When embedded in
+ * 3-D space, the isentropic vortex is still a solution to the corresponding 3-D
+ * system if the velocity along the third axis is a constant. In Cartesian
+ * coordinates \f$(x, y, z)\f$, and using dimensionless units, the primitive
+ * quantities at a given time \f$t\f$ are then
+ *
+ * \f{align*}
+ * \rho &= \left[1 - \dfrac{(\gamma - 1)\beta^2}{8\gamma\pi^2}\exp\left(
+ * 1 - r^2\right)\right]^{1/(\gamma - 1)}, \\
+ * v_x &= U - \dfrac{\beta\tilde y}{2\pi}\exp\left(\dfrac{1 - r^2}{2}\right),\\
+ * v_y &= V + \dfrac{\beta\tilde x}{2\pi}\exp\left(\dfrac{1 - r^2}{2}\right),\\
+ * v_z &= W,\\
+ * \epsilon &= \frac{\rho^{\gamma - 1}}{\gamma - 1},
+ * \f}
+ *
+ * with
+ *
+ * \f{align*}
+ * r^2 &= {\tilde x}^2 + {\tilde y}^2,\\
+ * \tilde x &= x - X_0 - U t,\\
+ * \tilde y &= y - Y_0 - V t,
+ * \f}
+ *
+ * where \f$(X_0, Y_0)\f$ is the position of the vortex on the \f$(x, y)\f$
+ * plane at \f$t = 0\f$, \f$(U, V, W)\f$ are the components of the mean flow
+ * velocity, \f$\beta\f$ is the vortex strength, and \f$\gamma\f$ is the
+ * adiabatic index. The pressure \f$p\f$ is then obtained from the dimensionless
+ * polytropic relation
+ *
+ * \f{align*}
+ * p = \rho^\gamma.
+ * \f}
+ *
+ * On the other hand, if the velocity along the \f$z-\f$axis is not a constant
+ * but a function of the \f$z\f$ coordinate, the resulting modified isentropic
+ * vortex is still a solution to the Newtonian Euler system, but with source
+ * terms that are proportional to \f$dv_z/dz\f$ (See
+ * NewtonianEuler::Sources::IsentropicVortexSource.)
+ *
+ * [1] H. Yee, N. Sandham, M. Djomehri, Low-dissipative high-order
+ * shock-capturing methods using characteristic-based filters, J. Comput. Phys.
+ * 150 (1999) 199, http://dx.doi.org/10.1006/jcph.1998.6177
+ */
+class IsentropicVortex {
+ public:
+  struct AdiabaticIndex {
+    using type = double;
+    static constexpr OptionString help = {"The adiabatic index of the fluid."};
+    // Note: bounds only valid for an ideal gas.
+    static type lower_bound() { return 1.0; }
+    static type upper_bound() { return 2.0; }
+    static type default_value() { return 5.0 / 3.0; }
```

Comment:
Do we want a default value for this? it doesn't seem like 5/3 is obviously the "default" value for the index